cmd/kops/toolbox_template.go

```diff
@@ -223,15 +223,47 @@ func newTemplateContext(files []string) (map[string]interface{}, error) {
 				return nil, fmt.Errorf("unable decode the configuration file: %s, error: %v", j, err)
 			}
 
-			for k, v := range ctx {
-				context[k] = v
-			}
+			context = mergeValues(context, ctx)
 		}
 	}
 
 	return context, nil
 }
 
+// Merges source and destination map, preferring values from the source map
+// Copied from the Helm (https://github.com/kubernetes/helm) project:
+// https://github.com/kubernetes/helm/blob/282984e75fd115a0765730efe09d8257c72fa56d/cmd/helm/install.go#L302
+func mergeValues(dest map[string]interface{}, src map[string]interface{}) map[string]interface{} {
+	for k, v := range src {
+		// If the key doesn't exist already, then just set the key to that value
+		if _, exists := dest[k]; !exists {
+			dest[k] = v
+			continue
+		}
+		nextMap, ok := v.(map[string]interface{})
+		// If it isn't another map, overwrite the value
+		if !ok {
+			dest[k] = v
+			continue
+		}
+		// If the key doesn't exist already, then just set the key to that value
+		if _, exists := dest[k]; !exists {
```

Comment:
Isn't this a duplicate of the first `if`?

Reply:
Good catch, I'll also make a PR for the Helm project